### PR TITLE
Move composer install into Makescript

### PR DESCRIPTION
**Pull request by Gomez**
Now we have the complete dev-setup in the Makefile:

Just do:
```
sudo install node nodejs-legacy curl
make all
```
@ChristophWurst We have to document that debug mode needs to be on, for js development, right?

### CODE REVIEW

**Comment by ChristophWurst** [comment]
this basically fixes #981

File: Makefile [hunk 1]
```diff
@@ -12,7 +12,11 @@ all: dist
 clean:
 	rm -rf $(build_dir)
 
-dist: install-npm-deps install-bower-deps optimize-js
+dist: install-composer-deps install-npm-deps install-bower-deps optimize-js
```
Comment (ChristophWurst) [inline Makefile]: please rename ``dist`` to ``install-deps`` and remove the ``optimize-js`` rule from there

**Comment by ChristophWurst** [comment]
From my understanding, the Makefile's only use case is packaging the app for the appstore/nightlies. Therefore, I'd say the default target should be ``appstore``

**Comment by ChristophWurst** [comment]
Additionally, the ``appstore`` target should depend on ``clean install-deps optimize-js``

**Comment by ChristophWurst** [comment]
We also have to update files excluded from the archive (bower.json, .bowerrc, translation-extractor.php,…). This can either be done in this PR or we create another once this one is merged.

**Comment by Gomez** [comment]
I would use the Makefile for all kind of "destinations", like:
make release (==appstore)
make dev-setup (with phantomsjs ...)
make (default == nightlies)

Would use save many troubles with the different setups...

**Comment by ChristophWurst** [comment]
> I would use the Makefile for all kind of "destinations", like:
> make release (==appstore)
> make dev-setup (with phantomsjs ...)
> make (default == nightlies)
> 
> Would use save many troubles with the different setups...

I'm not sure if that's a good idea. We would have to rely on ``make`` being available on every developer's system. Not all distros ship it by default…

There shouldn't be a difference between packaging nightlies and packaging for the appstore, right? :)

**Comment by ChristophWurst** [comment]
I'm closing this for now because we already have that in our current Makefile.

**Comment by ChristophWurst** [comment]
@Gomez do we need this branch or can we delete it?

**Comment by Gomez** [comment]
No need. Deleted

**Pull request closed without merging**